### DISCUSSION TITLE
Webhook (Laravel Nutgram)

### DISCUSSION THREAD

**Discussion by OlyaK95** [nutgram/nutgram#761] [Q&A]
Hi!

I have three files:

telegram.php
```
<?php
/** @var SergiX44\Nutgram\Nutgram $bot */

use SergiX44\Nutgram\Nutgram;

/*
|--------------------------------------------------------------------------
| Nutgram Handlers
|--------------------------------------------------------------------------
|
| Here is where you can register telegram handlers for Nutgram. These
| handlers are loaded by the NutgramServiceProvider. Enjoy!
|
*/

$bot->onCommand("start", function (Nutgram $bot) {
    return $bot->sendMessage("Hello, world!");
})->description("The start command!");
```
api.php
```
<?php
use Illuminate\Http\Request;
use Illuminate\Support\Facades\Route;
use App\Http\Controllers\FrontController;

Route::get("/user", function (Request $request) {
    return $request->user();
})->middleware("auth:sanctum");

Route::post("/webhook", FrontController::class);
```

FrontController.php
```
<?php

namespace App\Http\Controllers;

use Illuminate\Http\Request;
use Illuminate\Support\Facades\Log;
use SergiX44\Nutgram\Nutgram;

class FrontController extends Controller
{
    /**
     * Handle the telegram webhook request.
     */
    public function __invoke(Nutgram $bot)
    {
        Log::channel("telegram")->info("Webhook received", [
            "xyz" => "123",
            "data" => request()->all(),
        ]);

        $bot->run();
    }
}
```

When I run command `php artisan nutgram:run` and then in bot `/start`, I have `Hello, World!` message.
When I set webhook and then run bot `/start`, I have only Log message.

Why I don't receive `Hello, World!` on webhook?

Where to store the code so that it works for both Polling and Webhook modes?

**Comment by Lukasss93** [comment]
Did you set the webhook using the `php artisan nutgram:hook:set your-full-url-to-hook-route-here` command?

**Reply by OlyaK95** [reply]
> Did you set the webhook using the `php artisan nutgram:hook:set your-full-url-to-hook-route-here` command?

Yes, of course. I set url with this command and after no bot answer I try command `php artisan nutgram:hook:info` - no errors.

**Reply by OlyaK95** [reply]
> Did you set the webhook using the `php artisan nutgram:hook:set your-full-url-to-hook-route-here` command?

And I repeat that I get a log in the chat on webhook.
